src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -398,8 +398,9 @@ private bool StartCore(ProcessStartInfo startInfo)
             // Unix applications expect the terminal to be in an echoing state by default.
             // To support processes that interact with the terminal (e.g. 'vi'), we need to configure the
             // terminal to echo. We keep this configuration as long as there are children possibly using the terminal.
-            // We consider the child to be interactively using the terminal when both stdin and stdout are connected.
-            bool usesTerminal = !startInfo.RedirectStandardInput && !startInfo.RedirectStandardOutput;
+            bool usesTerminal = !(startInfo.RedirectStandardInput &&
+                                  startInfo.RedirectStandardOutput &&
+                                  startInfo.RedirectStandardError);
```

Comment:
for reference, discussion whether we should include RedirectStandardError here: https://github.com/dotnet/corefx/commit/4b35a6fe24aed9ba44ef2534bd32b7a1e0bb7841#r34854614